libtransmission/port-forwarding-natpmp.cc

```diff
@@ -140,16 +140,9 @@ tr_natpmp::PulseResult tr_natpmp::pulse(tr_port local_port, bool is_enabled)
         }
     }
 
-    if (state_ == State::Idle)
+    if (state_ <= State::Err && (is_mapped_ ? tr_time() >= renew_time_ : is_enabled && has_discovered_))
```

Comment:
For maintainability, enumerations shouldn't have ordered comparisons (since new enumeration values could be inserted or re-ordered, causing side-effects). Also that makes it harder to read/review for humans, since we don't know the affected enumeration values without looking them up.

I propose:
```suggestion
    if ((state_ == State::Idle || state_ == State::Err) && (is_mapped_ ? tr_time() >= renew_time_ : is_enabled && has_discovered_))
```

Reply:
6612dd5